src/TrackPanel.cpp

```diff
@@ -350,18 +350,13 @@ AudacityProject * TrackPanel::GetProject() const
    //Do it in two stages in case 'this' is not a wxWindow.
    //when the compiler will flag the error.
    wxWindow const * const pConstWind = this;
-   wxWindow * pWind=(wxWindow*)pConstWind;
-#ifdef EXPERIMENTAL_NOTEBOOK
-   pWind = pWind->GetParent(); //Page
-   wxASSERT( pWind );
-   pWind = pWind->GetParent(); //Notebook
-   wxASSERT( pWind );
-#endif
-   pWind = pWind->GetParent(); //MainPanel
-   wxASSERT( pWind );
-   pWind = pWind->GetParent(); //ProjectWindow
-   wxASSERT( pWind );
-   return &static_cast<ProjectWindow*>( pWind )->GetProject();
+   auto pWind=(wxWindow*)pConstWind;
+   while(pWind = pWind->GetParent())
```

Comment:
Put extra parentheses around this condition to suppress warnings (from clang at least).  Or rewrite as a for loop.